src/transformers/trainer.py

```diff
@@ -417,6 +417,42 @@ def __init__(
             elif FSDPOption.NO_SHARD in args.fsdp:
                 self.fsdp = ShardingStrategy.NO_SHARD
 
+        if args.xla_fsdp:
+            try:
+                from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as FSDP, checkpoint_module
+            except ImportError:
+                assert False, "Missing module XLAFullyShardedDataParallel; this module is available in torch-xla >= 1.12.0."
+            fsdp_kwargs = args.xla_fsdp_config
+            fsdp_wrap = lambda m: FSDP(m, **fsdp_kwargs)
```

Comment:
Let's not refactor in a lambda this for 2 uses, especially since it's so simple. Let's call FSDP with the kwargs in the 2 instances below instead.